wdl2cwl/main_oop_2.py

```diff
@@ -0,0 +1,250 @@
+            cwlVersion="v1.2",
+            baseCommand=base_command,
+        )
+
+        yaml = YAML()
+        yaml.default_flow_style = False
+        yaml.indent = 4
+        yaml.block_seq_indent = 2
+        result_stream = StringIO()
+        cwl_result = cat_tool.save()
+        scalarstring.walk_tree(cwl_result)
+        yaml.dump(cwl_result, result_stream)
+        yaml.dump(cwl_result, sys.stdout)
+
+        return result_stream.getvalue()
+
+    def get_cwl_docker_requirements(self, wdl_docker: WDL.Expr):
+        requirements: List[cwl.ProcessRequirement] = []
+        dockerpull = wdl_docker.expr.referee.expr.literal.value
+        requirements.append(cwl.DockerRequirement(dockerPull=dockerpull))
+        return requirements
+
+    def get_cwl_command_requirements(self, wdl_commands: List[str]) -> List[cwl.InitialWorkDirRequirement]:
+        """Translate WDL commands into CWL Initial WorkDir REquirement."""
+        command_str: str = ""
+        for wdl_command in wdl_commands:
+            if isinstance(wdl_command, str):
+                command_str += self.translate_wdl_str(wdl_command)
+            elif isinstance(wdl_command, WDL.Expr.Placeholder):
+                command_str += self.translate_wdl_placeholder(wdl_command)
+        return [
+            cwl.InitialWorkDirRequirement(
+                listing=[cwl.Dirent(entry=command_str, entryname="example.sh")]
+            )
+        ]
+    def translate_wdl_placeholder(self, wdl_placeholder) -> str:
+        cwl_command_str = ""
+        if isinstance(wdl_placeholder.expr, WDL.Expr.Get):
+            placeholder_name = wdl_placeholder.expr.expr.name
+            if wdl_placeholder.options:
+                if "true" in wdl_placeholder.options:
+                    cwl_command_str = (
+            "$(inputs."
+            + placeholder_name
+            + " ? "
+            + '"{true_value}" : "{false_value}")'.format(
+                true_value=wdl_placeholder.options["true"],
+                false_value=wdl_placeholder.options["false"],
+            )
+            + ")"
+        )
+            else:
+                cwl_command_str = "$(inputs." + wdl_placeholder.expr.expr.name + ")"
+        elif isinstance(wdl_placeholder.expr, WDL.Expr.Apply):
+            if len(wdl_placeholder.expr.arguments) == 1:
+                arg_name = wdl_placeholder.expr.arguments[0].expr.referee.name
+                cwl_command_str = (
+                    "$(inputs."
+                    + arg_name
+                    + " ? "
+                    + '"{false_value}" : "{true_value}")'.format(
+                        true_value=wdl_placeholder.options["true"],
+                        false_value=wdl_placeholder.options["false"],
+                    )
+                )
+            if len(wdl_placeholder.expr.arguments) == 2:
+                arg_name, arg_value = wdl_placeholder.expr.arguments
+                arg_name, arg_value = arg_name.literal.value, arg_value.expr.name
```

Comment:
I think we are creating `arg_name` and `arg_value` twice above :point_up:

Reply:
Am just reusing the variable names from line 133.